cli/src/main/java/org/jboss/as/cli/embedded/StopEmbeddedHostControllerHandler.java

```diff
@@ -52,7 +52,7 @@ public boolean isAvailable(CommandContext ctx) {
     @Override
     protected void doHandle(CommandContext ctx) throws CommandLineException {
         EmbeddedServerLaunch hostControllerLaunch = hostControllerReference.get();
-        if (hostControllerLaunch != null) {
+        if (hostControllerLaunch != null && hostControllerLaunch.getHostController() != null) {
```

Comment:
This is calling EmbeddedServerLaunch.getHostController() which in turn looks at EmbeddedServerReference to see if its a running hostcontroller or a standalone server. If this is a standalone server here, then we just ignore the shutdown, as standalone handles it instead.

As I recall it was a lot cleaner to implement it this way from the CLI side, but the issue was that you could use either shutdown command to shutdown either embedded process type.